Libraries/libutil/Sources/MemoryFilesystem.cpp

```diff
@@ -298,37 +264,140 @@ removeFile(std::string const &path)
     });
 }
 
-std::string MemoryFilesystem::
-resolvePath(std::string const &path) const
+bool MemoryFilesystem::
+isSymbolicLink(std::string const &path) const
 {
-    if (exists(path)) {
-        return FSUtil::NormalizePath(path);
-    } else {
-        return std::string();
-    }
+    return false;
+}
+
+ext::optional<std::string> MemoryFilesystem::
+readSymbolicLink(std::string const &path) const
+{
+    return ext::nullopt;
 }
 
 bool MemoryFilesystem::
-enumerateDirectory(
-    std::string const &path,
-    std::function<void(std::string const &)> const &cb) const
+writeSymbolicLink(std::string const &target, std::string const &path)
 {
-    return WalkPath<MemoryFilesystem::Entry const>(this, path, false, [&](MemoryFilesystem::Entry const *parent, std::string const &name, MemoryFilesystem::Entry const *entry) -> MemoryFilesystem::Entry const * {
+    return false;
+}
+
+bool MemoryFilesystem::
+copySymbolicLink(std::string const &from, std::string const &to)
+{
+    return false;
+}
+
+bool MemoryFilesystem::
+removeSymbolicLink(std::string const &path)
+{
+    return false;
+}
+
+bool MemoryFilesystem::
+isDirectory(std::string const &path) const
+{
+    return WalkPath<MemoryFilesystem::Entry const>(this, path, false, [](MemoryFilesystem::Entry const *parent, std::string const &name, MemoryFilesystem::Entry const *entry) -> MemoryFilesystem::Entry const * {
+        if (entry != nullptr && entry->type() != MemoryFilesystem::Entry::Type::Directory) {
```

Comment:
Nit:  You could unify isDirectory() and isFile() with a helper function that takes a MemoryFilesystem::Entry::Type.